validation/scripts/test_discovery.py

```diff
@@ -39,33 +42,65 @@ def get_test_data():
 
     for test in tests:
         # layers/data tuples
-        layers, data = get_data(test)
+        layers_file, data_file = get_data(test)
 
-        slabs = np.loadtxt(layers)
+        slabs = np.loadtxt(os.path.join(PTH, layers_file))
         assert slabs.shape[1] == 4
 
-        data = np.loadtxt(data)
+        data = np.loadtxt(os.path.join(PTH, data_file))
         assert data.shape[1] in [2, 3, 4]
 
         yield os.path.basename(test), slabs, data
 
 
 def get_data(file):
-    # for each of the unpolarised test files figure out where the data and
-    # layer parameters are
-
-    with open(file, "r") as f:
-        # ignore comment lines starting with # or space
-        while True:
-            line = f.readline()
-            if line.lstrip(" \t").startswith("#"):
-                continue
-            elif line == "\n":
-                continue
-            else:
-                layers = line.rstrip("\n")
-                data = f.readline().rstrip("\n")
-
-                layers = os.path.join(PTH, layers)
-                data = os.path.join(PTH, data)
-                return (layers, data)
+    # for each of the test files extract the parameters,
+    # e.g. the names of the data_file and layers_file
+
+    with open(file, "rt") as f:
+        # ignore comment lines starting with #
+        for line in f:
+            line = line.strip()
+            if not line.startswith("#"):
+                yield line
+
+
+def get_polarised_test_data():
+    """
+    A generator yielding (slabs, data) tuples.
+
+    `slabs` are np.ndarrays that specify the layer structure of the test.
+    ``slabs.shape = (N + 2, 6)``, where N is the number of layers.
+
+    The layer specification file has the layout:
+
+    # { "AGUIDE": angle_between_Q_and_H }
+    ignored     SLDn_fronting ignored      ignored   ignored ignored
+    thickness_1 SLDn_1        iSLD_1       thetaM_1  SLDm_1  rough_fronting1
+    thickness_2 SLDn_2        iSLD_2       thetaM_2  SLDm_2  rough_12
+    ignored     SLDn_backing  iSLD_backing ignored   ignored rough_backing2
+
+    `data` contains the test reflectivity data. It's an np.ndarray with
```

Comment:
No column for resolution smearing info here. I wonder if it suffices to just check resolution smearing in the unpolarised validation? Do most packages use the same machinery for calculating smearing? I'm not necessarily sure that that would always be the case.

Reply:
I agree this could be tricky; if there is for instance a sharp feature in one spin state but not the other, and your polarisation correction is not perfect, you would want to smear some of the "leaked" feature from other spin states.

I don't believe any of the analysis packages treat the spin states a mixtures, though - I think the assumption is always that the reported spin state is accurate and wholly correct (corrected).  In that case smearing is just the same as in the unpolarised case.  It's worth checking though.

We've certainly discussed the option of just storing the uncorrected spin states and doing the mixing in the model - there are some advantages to doing so.  But so far it is considered "best practice" at NIST to polarisation-correct the data and fit that, perhaps largely because it's easier to explain in a paper.